ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Types.hs

```diff
@@ -536,6 +559,24 @@ data PeerSelectionState peeraddr peerconn = PeerSelectionState {
      }
   deriving Show
 
+-- | A node is classified as a non-promiscuous node if it is a hidden relay or
+-- a BPa, e.g. if it is configured such that it can only have a chance to be
+-- connected to local roots. This is true if the node is configured in one of
+-- two ways:
+--
+-- * `DontUseBootstrapPeers`, `DontUseLedgerPeers` and
+--   `PeerSharingDisabled`; or
+-- * `UseBootstrapPeers`, `DontUseLedgerPeers` and
+--   `PeerSharingDisabled`, but it's not using any bootstrap peers (i.e. it is
+--   synced).
+--
+-- See `Ouroboros.Network.PeerSelection.Governor.readPromiscuousMode`.
+--
+data PromiscuousMode =
+     NonPromiscuousMode
+   | PromiscuousMode
+  deriving Show
```

Comment:
I think it would be worth to make even further distinctions here and explain what's classified as promiscuous or not depending on bootstrap/praos/genesis mode.

Reply:
but it doesn't depend on `praos / genesis` mode, or maybe I just don't understand your comment.

Reply:
Yes, I was thinking about `UntrustedState` and `TrustedStateWithExternalPeers`

Reply:
I am confused. It's the other way around. `OutboundConnectionsState` depends on `AssociationMode`, so we should rather expand the docs of `OutboundConenctionsState` if needed. Do you have any suggestions?